### PR TITLE
expression: implement vectorized evaluation for 'builtinLastDaySig'

### PR DESCRIPTION
**Pull request by myz000**
### What problem does this PR solve? 

expression: implement vectorized evaluation for 'builtinLastDaySig'

### What is changed and how it works?

goos: windows
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-2           1000000000               0.0610 ns/op          0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc-2                    1000000000               0.0350 ns/op          0 B/op          0 allocs/op
PASS

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -716,11 +716,35 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
...
 }
 
 func (b *builtinLastDaySig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: 1. We can use result to get the son's result directly without alloc a buffer.
2. We need to call b.arg[0].VecEvalTime to get son's result.

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -716,11 +716,35 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
...
+	ds := buf.Times()
+	result.ResizeTime(n, false)
+	times := result.Times()
+	for i := 0; i < n; i++ {
+		tm := ds[i].Time
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: We need check Null before calculate.

File: expression/builtin_time_vec.go [hunk 3]
```diff
@@ -716,11 +716,35 @@ func (b *builtinGetFormatSig) getFormat(format, location string) string {
...
+		year, month, day := tm.Year(), tm.Month(), tm.Day()
+		if month == 0 || day == 0 {
+			return handleInvalidTimeError(b.ctx, types.ErrIncorrectDatetimeValue.GenWithStackByArgs(buf.GetString(i)))
```
Comment (wshwsh12) [inline expression/builtin_time_vec.go]: If the error can be handled, we should set null and continue.

**Comment by ekalinin** [comment]
I'm sorry, but this function is vectorized by #13503

**Pull request closed without merging**